docs/getting_started.rst

```diff
@@ -289,258 +266,255 @@ Setting an attribute follows the ordinary convention for assigning attribute val
+Properties such as ``types`` and ``roles`` behave like Python lists, and list operations like ``append`` and ``extend`` will work directly on these kind of attributes:
 
 .. code:: python
 
-    >>> plasmid.roles = [ SO_PLASMID ]
-    >>> plasmid.roles = plasmid.roles + [ SO_CIRCULAR ]
+    >>> plasmid.roles = [ sbol3.SO_DOUBLE_STRANDED ]
+    >>> plasmid.roles.append( sbol3.SO_CIRCULAR )
+    
+    >>> plasmid.roles = []
+    >>> plasmid.roles.extend( [sbol3.SO_DOUBLE_STRANDED, sbol3.SO_CIRCULAR] )
+    
+    >>> plasmid.roles = [ sbol3.SO_DOUBLE_STRANDED ]
+    >>> plasmid.roles += [ sbol3.SO_CIRCULAR ]
 
 .. end
 
-To clear all values from an attribute, set to None:
+To clear all values from an attribute, set it to an empty list:
 
 .. code:: python
 
-    >>> plasmid.roles = None
+    >>> plasmid.roles = []
 
 .. end
 
 ------------------------------------------
-Creating, Adding and Getting Child Objects
+Creating and Adding Child Objects
 ------------------------------------------
 
-Some SBOL objects can be composed into hierarchical parent-child relationships.  In the specification diagrams, these relationships are indicated by black diamond arrows.  In the UML diagram above, the black diamond indicates that ComponentDefinitions are parents of SequenceAnnotations.  Properties of this type can be modified using the add method and passing the child object as the argument.
+Some SBOL objects can be composed into hierarchical parent-child relationships.  
+In the specification diagrams, these relationships are indicated by black diamond arrows.  
+In the UML diagram above, the black diamond indicates that Components are parents of Features.  
+In pySBOL3, properties of this type are created as subcomponents and then added to the appropriate 
+list attribute of the parent component. The constructor for the ``SubComponent`` class 
+takes a ``Component`` as its only required argument. In this usage, the ``Component`` is "... analogous 
+to a blueprint or specification sheet for a biological part..." Whereas the ``SubComponent`` "... represents 
+the specific occurrence of a part..." within a larger design 
+(`SBOL version 3.0.0 specification document <https://sbolstandard.org/docs/SBOL3.0specification.pdf>`_). 
+For example, to add a promoter to a circuit design, first define the promoter and circuit as SBOL 
+``Component`` objects, then define a ``SubComponent`` as an instance of the promoter and add that 
+``SubComponent`` to the circuit's ``features`` attribute:
 
 .. code:: python
 
-    >>> point_mutation = SequenceAnnotation('PointMutation')
-    >>> target_promoter.sequenceAnnotations.add(point_mutation)
+    >>> ptet = sbol3.Component('pTetR', sbol3.SBO_DNA)
+    >>> ptet.roles = [sbol3.SO_PROMOTER]
```

Comment:
Set `ptet.roles` and `circuit.roles` in the constructor, as an example of good practice.

Reply:
```suggestion
```